src/QMCHamiltonians/EwaldRef.h

```diff
@@ -43,7 +43,7 @@ enum
 /// Type for integers
 using IntType     = int;
 /// Type for floating point numbers
-using RealType    = QMCTraits::RealType;
+using RealType    = QMCTraits::FullPrecRealType;
```

Comment:
I think this is a very confusing alias. Please use FullPrecRealType or FPRT if you prefer something short.

Reply:
I'm tempted to just use `double` so that EwaldRef is fully decoupled from the rest of the QMCPACK configuration.

If you prefer, I'll just use `int_t` and `real_t` to avoid overlap with the naming of the normal hierarchy of QMCPACK types.

Reply:
That is better. Please do so. Because it is in the header and namespace, it can propagate in principle.

Reply:
I like the logic of the non-overlap / independent route as well.

Reply:
Done.